system_modules/napaudio/src/audio/service/audioservice.cpp

```diff
@@ -493,108 +66,7 @@ namespace nap
-			const PaDeviceInfo* outputDeviceInfo = Pa_GetDeviceInfo(outputDeviceIndex);
-			
-			if (inputDeviceInfo == nullptr)
-			{
-				// There is no input device
-				if (settings.mDisableInput == false && settings.mInputChannelCount > 0)
-				{
-					// Input channels were requested
-					if (configuration->mAllowChannelCountFailure)
-					{
-						Logger::warn("AudioService: input device not found, initializing without input channels.");
-						inputChannelCount = 0;
-					}
-					else {
-						errorState.fail("AudioService: input device not found.");
-						return false;
-					}
-				}
-				else
-					inputChannelCount = 0;
-			}
-			else {
-				// There is an input device
-				if (settings.mInputChannelCount > inputDeviceInfo->maxInputChannels)
-				{
-					// There are less channels than requested
-					if (configuration->mAllowChannelCountFailure)
-					{
-						Logger::warn(
-								"AudioService: Requested number of %i input channels not available, initializing with only %i",
-                                settings.mInputChannelCount, inputDeviceInfo->maxInputChannels);
-						inputChannelCount = inputDeviceInfo->maxInputChannels;
-					}
-					else {
-						errorState.fail("AudioService: Not enough available input channels on chosen device.");
-						return false;
-					}
-				}
-				else
-					// There are enough channels
-					inputChannelCount = settings.mInputChannelCount;
-			}
-			
-			
-			if (!outputDeviceInfo)
-			{
-				// There is no output device
-				if (settings.mDisableOutput == false && settings.mOutputChannelCount > 0)
-				{
-					if (configuration->mAllowChannelCountFailure)
-					{
-						Logger::warn("AudioService: output device not found, initializing without output channels.");
-						outputChannelCount = 0;
-					}
-					else {
-						errorState.fail("AudioService: output device not found.");
-						return false;
-					}
-				}
-				else
-					outputChannelCount = 0;
-			}
-			else {
-				// There is an output device
-				if (settings.mOutputChannelCount > outputDeviceInfo->maxOutputChannels)
-				{
-					// There are less channels than requested
-					if (configuration->mAllowChannelCountFailure)
-					{
-						Logger::warn(
-								"AudioService: Requested number of %i output channels not available, initializing with only %i",
-                                settings.mOutputChannelCount, outputDeviceInfo->maxOutputChannels);
-						outputChannelCount = outputDeviceInfo->maxOutputChannels;
-					}
-					else {
-						errorState.fail("AudioService: Not enough available output channels on chosen device.");
-						return false;
-					}
-				}
-				else
-					// There are enough channels
-					outputChannelCount = settings.mOutputChannelCount;
-			}
 
-			if (inputChannelCount < 1 && outputChannelCount < 0)
-			{
-				errorState.fail("Cannot start audio stream with zero input and output channels.");
-				return false;
-			}
-
-			return true;
-		}
-		
-		
 		void AudioService::checkLockfreeTypes()
```

Comment:
Should we keep this in ?

Reply:
Actually, I will uncomment the tests and commit to see if they are supported on all platforms nowadays. If not I will just get rid of it.